packages/runtime/src/runtime/hooks/useNavigateModal.tsx

```diff
@@ -50,40 +65,29 @@ export const useNavigateModalScreen: EnsembleActionHook<
     }
   }, [isStringAction, action]);
 
-  useEffect(() => {
-    if (!action || !modalContext || isComplete !== false) {
-      return;
-    }
+  const navigateCommand = useCommandCallback(
+    (evalContext, ...args) => {
+      if (!action || !modalContext) return;
 
-    navigateModalScreen(
-      action,
-      modalContext,
-      applicationContext?.application,
-      evaluatedInputs.inputs,
-      title,
-      onDismissCallback,
-    );
-    setIsComplete(true);
-  }, [
-    action,
-    applicationContext?.application,
-    evaluatedInputs.inputs,
-    isComplete,
-    modalContext,
-    onDismissCallback,
-    title,
-  ]);
+      const context = merge({}, evalContext, args[0]);
 
-  return useMemo(() => {
-    const callback = (args: unknown): void => {
-      if (!action) {
-        return;
-      }
+      const evaluatedInputs =
+        !isString(action) && !isNil(action.inputs)
+          ? evaluateInputs(cloneDeep(action.inputs), screenModel, context)
+          : {};
 
-      setIsComplete(false);
-      setContext(args);
-    };
+      navigateModalScreen(
+        action,
+        modalContext,
+        applicationContext?.application,
+        evaluatedInputs,
+        title,
+        onDismissCallback,
+      );
+    },
+    { navigate },
+    [action, applicationContext?.application, onDismissCallback, screenModel],
+  );
 
-    return { callback };
-  }, [action]);
+  return useMemo(() => ({ callback: navigateCommand }), [navigateCommand]);
```

Comment:
I thought we agreed we would have the components memoize the callback so this is not needed?

Reply:
components memoize the callback but we need to return the memoized callback from each action also, that we did in all the actions when we worked on the method stale PR

Reply:
`useCommandCalllback` already returns a memoized callback so this is not necessary.

Reply:
@evshi you are right that `useCommandCalllback` already returns a useCallback, but we are returning new object from the action and to memoize that we need this, if you check all other actions of ensembleActions, we are doing same

` return { callback: navigateCommand };`

please refer useNavigateUrl, useExecuteCode, useCommandCallback, useCloseAllDialogs and so on

Reply:
@sagardspeed2 that is what my original comment is referring to. We discussed last week that it is not necessary to memoize the return object because the consuming component should only depend on the callback instead of the object.